firefox-ios/Client/Frontend/Library/HistoryPanel/HistoryPanelViewModel.swift

```diff
@@ -248,14 +248,33 @@ class HistoryPanelViewModel: FeatureFlaggable {
     }
 
     func deleteGroupsFor(dateOption: HistoryDeletionUtilityDateOptions) {
-        guard let deletableSections = getDeletableSection(for: dateOption) else { return }
+        if dateOption != .lastHour {
+            guard let deletableSections = getDeletableSection(for: dateOption) else { return }
+            deletableSections.forEach { section in
+                // Remove grouped items for delete section
+                var sectionItems: [AnyHashable] = groupsForSection(section: section)
+                let singleItems = groupedSites.itemsForSection(section.rawValue - 1)
+                sectionItems.append(contentsOf: singleItems)
+                removeHistoryItems(item: sectionItems, at: section.rawValue)
+            }
```

Comment:
we could probably throw this into it's own function and make the parent function super easy for the readings